src/Controller/CRUDController.php

```diff
@@ -1069,7 +1081,8 @@ protected function handleModelManagerException(\Exception $exception): void
      *
      * @throws ModelManagerThrowable
      *
-     * @return string|null A custom error message to display in the flag bag instead of the generic one
+     * @return Response|string|null A custom response to redirect to, or a custom error message to
```

Comment:
This looks very hacky and violates SRP.

It would be better to deprecate this method and replace it with events.

Reply:
Sure, but if you just dispatch an event when an exception occur: 
- How do you display another message than the default one from sonata in the flashbag ?
- How do you display a whole different response than the one the controller will display ?

I'll be happy to introduce event then

Reply:
You could register a default event listener for the flashbag handling and intercept the event using the [`stopPropagation` ](https://github.com/symfony/symfony/blob/3be9bfb3e6141b9024bcc01006e8b51d9391b11a/src/Symfony/Contracts/EventDispatcher/Event.php#L50) method